source/common/stats/thread_local_store.cc

```diff
@@ -152,7 +152,59 @@ ScopeSharedPtr ThreadLocalStoreImpl::createScope(const std::string& name) {
   return scopeFromStatName(stat_name_storage.statName());
 }
 
+class NullScope : public Scope {
```

Comment:
Curious to know how this will interact with the stats matcher. What will cause the creation of a `NullScope`

Reply:
This needs to be thought through better, but in this PR see the implementation of createScope -- it matches the scope name against the matcher.

I think it needs to be a little bit more careful about that. Like scope "foo" should match against matcher-prefix "foo." or regex "foo\..*" or similar, but actually not against literal "foo" or prefix "foo".

Also I'm wondering about matcher mutations. There is some handling of adding a stats matcher *after* having constructed some stats. I don't think there's any handling of mutating the matcher after it has been added.

Reply:
🤦 I missed that. Makes sense.